src/parser/hunter/shared/modules/talents/AMurderOfCrows.js

```diff
@@ -43,29 +40,32 @@ class AMurderOfCrows extends Analyzer {
       return;
     }
     this.casts++;
-    this.enemyID = encodeTargetString(event.targetID, event.targetInstance);
-    this.applicationTimestamp = event.timestamp;
+    this.applicationTimestamp = null;
     this.lastDamageTick = null;
   }
 
   on_byPlayer_damage(event) {
     const spellId = event.ability.guid;
-    if (this.lastDamageTick && event.timestamp + MS_BUFFER < this.applicationTimestamp + CROWS_DURATION && event.timestamp > this.lastDamageTick + CROWS_TICK_RATE + MS_BUFFER && this.spellUsable.isOnCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id)) {
+    /**
+     * Checks if we've had atleast 1 damage tick of the currently applied crows, and checks that crows is in fact on cooldown.
+     * Then it checks whether the current damage event is less than the full duration of crows, followed by a check to see if more than 1 second has passed since last tick
+     * If more than 1 second has passed, we can assume that crows has been reset, and thus we the CD.
+     */
+    if (this.lastDamageTick && this.spellUsable.isOnCooldown(SPELLS.A_MURDER_OF_CROWS_TALENT.id) && event.timestamp + MS_BUFFER < this.applicationTimestamp + CROWS_DURATION + MS_BUFFER && event.timestamp > this.lastDamageTick + CROWS_TICK_RATE + MS_BUFFER) {
```

Comment:
Would be easier to read if this was split into multiple lines

Reply:
`event.timestamp + MS_BUFFER < this.applicationTimestamp + CROWS_DURATION + MS_BUFFER`
`MS_BUFFER` is added on both sides of the less than here.

Reply:
Multilined the if statement and split the comments up to make it more readable (imo) and removed the 2x MS_BUFFER